include/libnuraft/internal_timer.hxx

```diff
@@ -122,10 +120,11 @@ struct timer_helper {
     }
 
     static uint64_t get_timeofday_us() {
-        struct timeval tv;
-        gettimeofday(&tv, nullptr);
-        uint64_t ret = tv.tv_sec * 1000000UL;
-        ret += tv.tv_usec;
+        namespace sc = std::chrono;
+        sc::system_clock::duration d = sc::system_clock::now().time_since_epoch();
+        sc::seconds s = sc::duration_cast<sc::seconds>(d);
+        uint64_t ret = s.count() * 1000000UL;
+        ret += sc::duration_cast<sc::microseconds>(d - s).count();
```

Comment:
Can we just do it as follows?
```c++
sc::microseconds s = sc::duration_cast<sc::microseconds>(d);
uint64_t ret = s.count();
```

Reply:
Yes, this Copy/Paste from other code that kept them separate. Will change.